### ISSUE TITLE
Wrong carousel size calculation formula when includeGapInSize = true

### ISSUE THREAD

**Issue by 7iomka** [mantinedev/mantine#2447]
### What package has an issue

@mantine/carousel

### Describe the bug

When `includeGapInSize` true, we have styles logic on item like that
```css
	flex: 0 0 calc(var(--slideWidth) - var(--slideGap) / 2);
	margin-right: var(--slideGap);
```

This produce wrong calculations
https://codepen.io/7iomka/pen/xxjRQGw

And get me unexpected behavior when I'm slide with nav buttons

https://user-images.githubusercontent.com/18473137/190236075-87fc1aa7-44e9-4f72-8bab-1740aa3a8d9f.mov

### What version of @mantine/hooks page do you have in package.json?

latest

### Do you know how to fix the issue

Yes

### Possible fix

Use another formula
One of popular solutions:
https://codepen.io/7iomka/pen/abGBQmd

**Comment by gyancodes** [comment]
Hey @rtivital  @7iomka can I work on this issue?

**Comment by rtivital** [comment]
Yes, sure

**Comment by gyancodes** [comment]
Hello, @7iomka. Could you please advise me on how to resolve this problem?

**Comment by 7iomka** [comment]
@gyancodes, @rtivital 
Because now we have not access to props like `includeGapInSize` or `slideGap`, `slideSize`, from `CarouselStylesParams` to setup this fix as config on provider, we need fix in the lib.

Sample code
```ts
  container: {
    marginLeft: -params.slideGap / 2,
    marginRight: -params.slideGap / 2,
  },
  slide: {
    marginLeft: params.slideGap / 2,
    marginRight: params.slideGap / 2,
    flex: `0 0 calc(${params.slideSize} - ${params.slideGap})`
  }
```
Of course, you need also to check somehow internally case when user not passed `slideGap` or `slideSize` but instead used props for `Carousel.Slide`, and, also if none of this passed to use fallback (default) values

**Comment by gyancodes** [comment]
@7iomka  How I can reproduce It, I am facing issues.

**Comment by 7iomka** [comment]
> @7iomka How I can reproduce It, I am facing issues.

I created a demo component which demonstrates the problem.
Use it somehow locally to make changes in the lib to fix formula.
https://codesandbox.io/p/github/7iomka/effector-react-form-flow/draft/suspicious-austin?file=%2Fsrc%2Fwidgets%2Fproduct%2Fproduct-carousel%2Fproduct-carousel.component.tsx&selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A20%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A19%7D%5D

**Comment by 7iomka** [comment]
@gyancodes Hi, any news on this?)

**Comment by gyancodes** [comment]
@7iomka Still I am working on it 🤦‍♂️, I think It will take some time.

**Comment by 7iomka** [comment]
> @7iomka Still I am working on it 🤦‍♂️, I think It will take some time.

Any progress? Tell me what difficulties you face? For me, this is a blocking bug that breaks the behavior of many carousels in my project

**Comment by gyancodes** [comment]
I am withdrawing from this issue as I am taking more time which will later create a problem. However, I will try to solve this asap.

Thanks and Regards

**Comment by 7iomka** [comment]
@gyancodes , @rtivital Hi guys
See the [demo](https://codesandbox.io/p/github/7iomka/effector-react-form-flow/draft/suspicious-austin?file=%2Fsrc%2Fwidgets%2Fproduct%2Fproduct-carousel%2Fproduct-carouse.styles.ts&selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A52%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A51%7D%5D)

I fixed the problem locally, please take a look.
Locally, we have the following restrictions, due to which this fix is only applicable under the condition that all props passed to the component must be passed to the hook (`includeGapInSize`, `orientation`, `breakpoints`, `slideGap`, `slideSize`), and props `gap` and `size` should not be set at the slide level, but to use `slideGap` and `slideSize` on carousel level:
- we can't use getRef in component settings at provider level
- mantine does not provide public access to carousel context

inside, you can solve this problem completely, since you have access to the context.
The main idea is
- `container`
  -   If `includeGapInSize` is true: negative margin with value of `slideGap` (right for horizontal orientation, bottom for vertical one)
  -   Else: nothing
- `slide` 
  -  If `includeGapInSize` is true: **padding** in the same direction and value as `slideGap` if `includeGapInSize` is true
  -  Else use margin instead
  -  `flex` value does not contain any subtractions and only represents direct `slideSize` value as flex-basis.

In short, we use `padding` in one direction with it value subtraction on container in case when we want to `includeGapInSize`, otherwise we just use margins.

Clean embla implementation demo for horizontal carousel can be found [here](https://codesandbox.io/s/embla-carousel-arrows-dots-react-forked-6kzu72?file=/src/css/embla.css)

It's all thanks to a fruitful [discussion](https://github.com/davidjerleke/embla-carousel/issues/376)

**Comment by rtivital** [comment]
@7iomka can you create a PR with a fix?

**Comment by 7iomka** [comment]
> @7iomka can you create a PR with a fix?

yes, give me rights, please
